tests/unit/verticals/secret/output/test_json_output.py

```diff
@@ -238,10 +353,22 @@ def check_occurrences_indices(occurrences: List[Dict[str, Any]], patch: str) ->
         "_SINGLE_ADD_PATCH",
         "_SINGLE_DELETE_PATCH",
         "_SINGLE_MOVE_PATCH",
+        "_MULTILINE_SECRET_FILE",
+        "_SINGLE_LINE_SECRET_FILE",
+        "_ONE_LINE_AND_MULTILINE_FILE",
     ],
 )
-def test_json_output(client, cache, name, input_patch, expected_exit_code):
-    c = Commit.from_patch(input_patch)
+def test_json_output_for_patch(
+    client, cache, name, input, expected_exit_code, is_patch, tmp_path
+):
+    if is_patch:
+        c = Commit.from_patch(input)
+    else:
+        test_file = tmp_path / "file"
+        with open(test_file, "w", newline="\n") as f:
+            f.write(input)
+        file = File(path=test_file)
```

Comment:
minor: you can avoid the `if is_patch` below by creating a scannable list in each `if` branch here. Something like this:

```python
    if is_patch:
        commit = Commit.from_patch(input)
        scannables = commit.get_files()
    else:
        test_file = tmp_path / "file"
        with open(test_file, "w", newline="\n") as f:
            f.write(input)
        scannables = [File(path=test_file)]
```

Then pass `scannables` to `scanner.scan()`.